DMCompiler/Compiler/DM/DMParser.cs

```diff
@@ -166,7 +168,24 @@ public DMASTStatement Statement(bool requireDelimiter = true)
                     if (Check(TokenType.DM_LeftParenthesis)) {
                         DMCompiler.VerbosePrint($"Parsing proc {_currentPath}()");
                         BracketWhitespace();
-                        DMASTDefinitionParameter[] parameters = DefinitionParameters();
+                        var parameters = DefinitionParameters();
+
+                        if (Current().Type != TokenType.DM_RightParenthesis && Current().Type != TokenType.DM_Comma &&
+                            !Check(TokenType.DM_IndeterminateArgs))
+                        {
+                            Error($"error: {parameters.Last().Name}: missing comma ',' or right-paren ')'", false);
```

Comment:
`parameters.Last()` throws an exception if there aren't any parameters.

For example: `proc/t(`